src/memory_service/fam_memory_service_server.cpp

```diff
@@ -287,6 +287,45 @@ Fam_Memory_Service_Server::copy(::grpc::ServerContext *context,
     // Return status OK
     return ::grpc::Status::OK;
 }
+::grpc::Status Fam_Memory_Service_Server::backup(
+    ::grpc::ServerContext *context,
+    const ::Fam_Memory_Backup_Restore_Request *request,
+    ::Fam_Memory_Backup_Restore_Response *response) {
+    MEMORY_SERVICE_SERVER_PROFILE_START_OPS()
+    try {
+        memoryService->backup(request->region_id(), request->offset(),
+                              request->filename(), request->size());
+
+    } catch (Memory_Service_Exception &e) {
+        response->set_errorcode(e.fam_error());
+        response->set_errormsg(e.fam_error_msg());
+        return ::grpc::Status::OK;
+    }
+
+    MEMORY_SERVICE_SERVER_PROFILE_END_OPS(mem_server_copy);
+    // Return status OK
+    return ::grpc::Status::OK;
+}
+
+::grpc::Status Fam_Memory_Service_Server::restore(
+    ::grpc::ServerContext *context,
+    const ::Fam_Memory_Backup_Restore_Request *request,
+    ::Fam_Memory_Backup_Restore_Response *response) {
+    MEMORY_SERVICE_SERVER_PROFILE_START_OPS()
+    try {
+        memoryService->restore(request->region_id(), request->offset(),
+                               request->filename(), request->size());
+
+    } catch (Memory_Service_Exception &e) {
+        response->set_errorcode(e.fam_error());
+        response->set_errormsg(e.fam_error_msg());
+        return ::grpc::Status::OK;
```

Comment:
MEMORY_SERVICE_SERVER_PROFILE_END_OPS is not set in case of an error. Also do try to compile/run tests with profiling enabled.